lib/danger/danger_core/plugins/dangerfile_bitbucket_server_plugin.rb

```diff
@@ -0,0 +1,36 @@
+# coding: utf-8
+require 'danger/plugin_support/plugin'
+
+module Danger
+  class DangerfileBitbucketServerPlugin < Plugin
+    def self.new(dangerfile)
+      return nil if dangerfile.env.request_source.class != Danger::RequestSources::BitbucketServer
+      super
+    end
+
+    def self.instance_name
+      "bitbucket_server"
```

Comment:
Unfortunately not. Bitbucket and Stash (Bitbucket Server) are two separate products and they have different APIs.
https://developer.atlassian.com/static/rest/stash/3.11.6/stash-rest.html vs https://confluence.atlassian.com/bitbucket/use-the-bitbucket-cloud-rest-apis-222724129.html